lib/Twig/Extension/Core.php

```diff
@@ -821,7 +821,27 @@ function _twig_default_filter($value, $default = '')
 function twig_get_array_keys_filter($array)
 {
     if ($array instanceof Traversable) {
-        return array_keys(iterator_to_array($array));
+        while ($array instanceof IteratorAggregate) {
+            $array = $array->getIterator();
+        }
+
+        if ($array instanceof Iterator) {
+            $keys = array();
+            $array->rewind();
+            while ($array->valid()) {
+                $keys[] = $array->key();
+                $array->next();
+            }
+
+            return $keys;
+        }
```

Comment:
I think iterators like that are becoming more and more popular because of the `Iterable` addition to PHP 7 (which is awesome btw. :) ). Iterators like this are very powerful and reduce a lot of boiler template code.

I'm not sure what OTR mean, but I think the RFC is nice.

Sorry @jbafford for pinging you here, I couldn't open an issue over @ https://github.com/jbafford/php-src/tree/foreachvoid to get in contact with you. Are you still working on the RFC or willing to pick it up again?